ocaml/xenopsd/xc/service.ml

```diff
@@ -235,6 +235,14 @@ module type DAEMONPIDPATH = sig
   val name : string
 
   val pid_location : Pid.path
+
+  (* To check if a pid is valid, look up the commandline args for the process
+     pid, check that expected args are included. The expected args are set by
+     (expected_cmdline_args domid) for each service.
+     Note that the parameter "domid" here is required as otherwise we can not
+     distinguish between the process instances of the same service for
+     different domains. *)
+  val expected_cmdline_args : int -> string list
```

Comment:
I would suggest to make this a named argument `domid` - there is potential to confuse this with a pid.